aws-s3/credentials.ml

```diff
@@ -74,19 +74,41 @@ module Make(Io : Types.Io) = struct
       let ini = new Inifiles.inifile creds_file in
       let access_key = ini#getval profile "aws_access_key_id" in
       let secret_key = ini#getval profile "aws_secret_access_key" in
-      make ~access_key ~secret_key () |> Deferred.Or_error.return
+      let token = match ini#getval profile "aws_session_token" with
+      | x -> Some x
+      | exception Inifiles.Invalid_element _ -> None
+      | exception Inifiles.Invalid_section _ -> None
+      in
+      make ~access_key ~secret_key ?token () |> Deferred.Or_error.return
+  end
+
+  module Env = struct
+    let get_credentials () =
+      let access_key = Sys.getenv_opt "AWS_ACCESS_KEY_ID" in
+      let secret_key = Sys.getenv_opt "AWS_SECRET_ACCESS_KEY" in
+      let token = Sys.getenv_opt "AWS_SESSION_TOKEN" in
+      match access_key, secret_key with
+      | None, _ -> None
+      | _, None -> None
+      | Some access_key, Some secret_key ->
+        Some (make ~access_key ~secret_key ?token () |> Deferred.Or_error.return)
   end
 
   module Helper = struct
```

Comment:
Any updates to the logic should be reflected in the documentation in the .mli file.